### PR TITLE
Csr feat/bugfix for rejected response callback invokation

### PR DESCRIPTION
**Pull request by aggarw13**
*Issue #, if available:*

*Description of changes:*
* Fix bug in Provisioning parser that was not calling user-callback for rejected server response case
* Update parser unit tests to check for callback invocations to prevent future un-detection

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: libraries/aws/provisioning/src/aws_iot_provisioning_parser.c [hunk 1]
```diff
@@ -120,88 +120,91 @@ static AwsIotProvisioningError_t _parseRejectedResponse( IotSerializerDecoderObj
...
     IotSerializerDecoderObject_t statusCodeDecoder = IOT_SERIALIZER_DECODER_OBJECT_INITIALIZER;
     IotSerializerDecoderObject_t errorCodeDecoder = IOT_SERIALIZER_DECODER_OBJECT_INITIALIZER;
     IotSerializerDecoderObject_t errorMessageDecoder = IOT_SERIALIZER_DECODER_OBJECT_INITIALIZER;
 
     /* Suppress warning when parameter is unused in non-Debug mode. */
     ( void ) pOperationName;
 
-    /* Initialize the status as "server refused" as we are parsing the "rejected" response. */
-    IOT_FUNCTION_ENTRY( AwsIotProvisioningError_t, AWS_IOT_PROVISIONING_SERVER_REFUSED );
+    AwsIotProvisioningError_t status = AWS_IOT_PROVISIONING_SUCCESS;
```
Comment (dan4thewin) [inline libraries/aws/provisioning/src/aws_iot_provisioning_parser.c]: What does it mean when a function named _parseRejectedResponse() returns provisioning success?  Does that mean parse success?

File: libraries/aws/provisioning/src/aws_iot_provisioning_parser.c [hunk 2]
```diff
@@ -406,8 +409,13 @@ AwsIotProvisioningError_t _AwsIotProvisioning_ParseKeysAndCertificateResponse( A
             /* Invoke the user-provided callback with the parsed rejected data, if parsing was successful . */
             if( status == AWS_IOT_PROVISIONING_SUCCESS )
             {
+                IotLogInfo( "parser: About to call user-callback with rejected server response: Operation={\"%s\"}",
```
Comment (dan4thewin) [inline libraries/aws/provisioning/src/aws_iot_provisioning_parser.c]: "About to call" -> "calling" ?

File: libraries/aws/provisioning/src/aws_iot_provisioning_parser.c (hunk 1 above)
Comment (aggarw13) [reply libraries/aws/provisioning/src/aws_iot_provisioning_parser.c]: Yes, as the goal of the function is to only parse the payload, my thought is to not add more responsibility/information in its return value than what its purpose warranties :)

Comment (dan4thewin) [reply libraries/aws/provisioning/src/aws_iot_provisioning_parser.c]: OK.  Maybe in the future we can make these constants less confusing.

**Pull request merged**